### PR TITLE
[PXN-2217](Fixed) - fix the infinite alloc/repeat of screen content each viewWillAppear() 

### PR DESCRIPTION
**Pull request by ricardo-dalambert-meli**
## What have changed?

Added a new recursive method for correctly clear all UIView tree hierarchy of a single UIview. Also added two calls for it in `setupScrollView()` and `renderContentView()` in order to correctly remove all old instances before replace them with the new updated ones.

## Kind of pr:

- [x] BugFix 
- [ ] Feature
- [ ] Improvement
- [ ] Refactor

## How to test:

Every time `viewWillAppear()` is called, a new set of UX objects is instanced over the previous ones. The change on `setupScrollView()` with a correct recursive removal of the view tree hierarchy solves the wrong stacking of `buildFooterView()`, and the second change is made to avoid the the "super-stack" behavior on 'renderContentView()' , where several groups of repeated content were available for scrolling in the UIScrollView.

## Prints for layout changes:

`buildFooterView()` stack and super-stack of content evidences attached on issue:

https://mercadolibre.atlassian.net/browse/PXN-2217

### CODE REVIEW

File: MercadoPagoSDK/MercadoPagoSDK/Helpers/UIView+Extensions.swift [hunk 1]
```diff
@@ -8,4 +8,14 @@ extension UIView {
...
+    func clearSubViews(_ currentView: UIView? = nil) {
+        let view = currentView != nil ? currentView : self
+        if view?.subviews.count ?? 0 > 0 {
+            for subView in view?.subviews ?? [] {
+                clearSubViews(subView)
+            }
+        }
+        view?.removeFromSuperview()
+    }
+
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/Helpers/UIView+Extensions.swift:20]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/1c527c7e083e4e7c22207b438a364c61f350a900/MercadoPagoSDK/MercadoPagoSDK/Helpers/UIView+Extensions.swift#L20"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>UIView+Extensions.swift:20</code></td>
    </tr>
  </tbody>
</table>

Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/Helpers/UIView+Extensions.swift:20]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/1c527c7e083e4e7c22207b438a364c61f350a900/MercadoPagoSDK/MercadoPagoSDK/Helpers/UIView+Extensions.swift#L20"></span>Don't include vertical whitespace (empty line) before closing braces.<br />
<code>vertical_whitespace_closing_braces</code> <code>UIView+Extensions.swift:20</code></td>
    </tr>
  </tbody>
</table>

File: MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift [hunk 2]
```diff
@@ -136,7 +135,9 @@ class PXNewResultViewController: MercadoPagoUIViewController {
     }
 
     private func renderContentView() {
-
+
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift:138]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/1c527c7e083e4e7c22207b438a364c61f350a900/MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift#L138"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>PXNewResultViewController.swift:138</code></td>
    </tr>
  </tbody>
</table>

Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift:138]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/1c527c7e083e4e7c22207b438a364c61f350a900/MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift#L138"></span>Don't include vertical whitespace (empty line) after opening braces.<br />
<code>vertical_whitespace_opening_braces</code> <code>PXNewResultViewController.swift:138</code></td>
    </tr>
  </tbody>
</table>

File: MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift [hunk 3]
```diff
@@ -136,7 +135,9 @@ class PXNewResultViewController: MercadoPagoUIViewController {
     }
 
     private func renderContentView() {
-
+        
+        contentView.clearSubViews()
+
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift:140]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/1c527c7e083e4e7c22207b438a364c61f350a900/MercadoPagoSDK/MercadoPagoSDK/UI/PXResult/New UI/PXNewResultViewController.swift#L140"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>PXNewResultViewController.swift:140</code></td>
    </tr>
  </tbody>
</table>

**Review by pesilsa — APPROVED** [review]

**Pull request merged**